.github/workflows/build-docker-image.yml

```diff
@@ -0,0 +1,67 @@
+name: Build Docker Image
+
+on:
+  workflow_call:
+    inputs:
+      tags:
+        required: true
+        type: string
+        description: "The tags to use for the Docker image"
+      target:
+        required: true
+        type: string
+        description: "The target to use for the Docker image"
+      build_args:
```

Comment:
Can't we use `vars` for some of these inputs?

>To reuse variables in multiple workflows, set them at the organization, repository, or environment levels and reference them using the vars context. For more information see "[Store information in variables](https://docs.github.com/en/actions/learn-github-actions/variables)" and "[Accessing contextual information about workflow runs](https://docs.github.com/en/actions/learn-github-actions/contexts#vars-context)."

Reply:
@kilemensi Since this is a reusable workflow, we need to [define what will be passed to it when it is called](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow). Also we can't use environment variables since they are not [passed down](https://docs.github.com/en/actions/sharing-automations/reusing-workflows#limitations) to the reusable workflow:
> Any environment variables set in an env context defined at the workflow level in the caller workflow are not propagated to the called workflow. For more information, see "[Store information in variables](https://docs.github.com/en/actions/learn-github-actions/variables)" and "[Accessing contextual information about workflow runs](https://docs.github.com/en/actions/learn-github-actions/contexts#env-context)."

Reply:
`env` and `vars` are separate @kelvinkipruto. Please read my comment again.

--

Adding this here for more context: https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#defining-configuration-variables-for-multiple-workflows